src/components/Dropdown/V2/Dropdown.MenuContainer.tsx

```diff
@@ -310,6 +310,15 @@ export class MenuContainer extends React.PureComponent<Props> {
     this.didOpen = false
     // End the reposition cycle
     cancelAnimationFrame(this.positionRAF)
+    this.props.closeDropdown()
+    this.focusTriggerNode()
+  }
+
+  focusTriggerNode = () => {
+    const { triggerNode } = this.props
```

Comment:
`PortalWrapper` does have a refocus trigger mechanism. However, since Dropdown doesn't set the trigger node in the standard `PortalWrapper` manner, we have to handle it ourselves.